### PR TITLE
Adding Data processing models

### PR DESCRIPTION
**Pull request by SnakyMz**
Things I have done:

- [ ] Update user_id to author_id in posts table
- [ ] Added User model with methods
- [ ] Added Post model with associations and methods
- [ ] Added Comment model with associations and methods
- [ ] Added Like model with associations and methods

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  has_many :likes
+
+  def update_posts_counter
+    author.increment!(:posts_counter)
+  end
+
+  def recent_comments
+    comments.order(created_at: :desc).limit(5)
+  end
+end
```
Comment (Munsa1) [inline app/models/post.rb]: - [ ] Kindly add a rails callback  using [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the `#update_posts_counter` method every time a new post has been saved to database, then make the `update_posts_counter` method private[Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - after_save to call the #update_post_counter method every time a new post has been saved to database, then make the update_post_counter method private

e.g

```ruby
after_save :update_posts_counter

  private

  def update_posts_counter
    author.increment!(:posts_counter)
  end
```


#### Take NOTE to only make the `update_posts_counter` method private ONLY

File: app/models/comment.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+class Comment < ApplicationRecord
+  belongs_to :user
+  belongs_to :post
+
+  def update_comments_counter
+    post.increment!(:comments_counter)
+  end
+end
```
Comment (Munsa1) [inline app/models/comment.rb]: - [ ] Again here just as we have to do in the post model, kindly add a rails callback  using [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the `#update_comments_counter` method every time a new comment has been saved to database, then make the `update_comments_counter` method private [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - after_save to call the #update_comments_counter method every time a new post has been saved to database

e.g

```ruby
after_save :update_comments_counter

 

  def update_comments_counter
    author.increment!(:comments_counter)
  end
```

File: app/models/like.rb [hunk 3]
```diff
@@ -0,0 +1,8 @@
+class Like < ApplicationRecord
+  belongs_to :user
+  belongs_to :post
+
+  def update_likes_counter
+    post.increment!(:likes_counter)
+  end
+end
```
Comment (Munsa1) [inline app/models/like.rb]: - [ ] Lastly again for the Like model kindly add the Rails Active Call Back as explained for the Comment model [here](https://github.com/SnakyMz/Blog-Ruby/pull/2#discussion_r1372139482)

**Review by Munsa1 — CHANGES_REQUESTED** [review]
Hi @SnakyMz, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
Clean CodeBase :heavy_check_mark: 
Passing Linters :heavy_check_mark: 
Correct Git Flow :heavy_check_mark: 
Descriptive ReadMe File :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/like.rb (hunk 3 above)
Comment (SnakyMz) [reply app/models/like.rb]: Thank you very much, I will make changes as suggested

**Review by Felix45 — APPROVED** [review]
# Approved :green_circle: 
Hi @SnakyMz, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights :heavy_check_mark: 
- [x] All models have been set correctly :white_check_mark: 
- [x] All migrations run as expected :white_check_mark: 
- [x] Pull request has a descriptive title and summary :white_check_mark: 
- [x] Repository has a professional readme file :white_check_mark:  
- [x] Implemented all the changes from the previous reviewer :white_check_mark:  

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**